gitgud/operations.py

```diff
@@ -155,6 +157,54 @@ def commit(self, commit_message, parents, time_offset):
                 skip_hooks=True)
         return commit_obj
 
+    def file_in_commit(self, commit, filepath):
+        commit = self.repo.commit(commit)
+        return filepath in commit.tree
+
+    def get_commit_content(self, commit, filepath):
+        if commit in self._streamed_content:
+            return self._streamed_content[commit]
+
+        commit = self.repo.commit(commit)
+
+        if not isinstance(filepath, str):
+            filepath = str(filepath)
+
+        commit_content = (commit.tree / filepath).data_stream.read().decode("ascii")  # noqa: E501
+        self._streamed_content[commit] = commit_content
+        return commit_content
+
+    def get_staging_area(self):
+        return self.get_tracked_changes("staging")
+
+    def get_working_area(self):
+        changes = self.get_tracked_changes("working")
+        changes["added"] += self.repo.untracked_files
+        return changes
+
+    def get_tracked_changes(self, change_type):
```

Comment:
When might this method be used? I think we're more interested in the content of the working directory or index, not the changes. 

We're definitely interested in the content. For example imagine there are two files, 1.txt and 2.txt. 1.txt is already committed and 2.txt has been staged. I would want a method `get_staging_area_content()` that returns a dict with the content, like this: `{"1.txt": "Hello world", "2.txt": "Is this content?"}`